config/config.go

```diff
@@ -183,28 +183,27 @@ func Load(
 	return &config, nil
 }
 
-// decodeTypeAndName decodes a key in type[/name] format into type and fullName.
+// DecodeTypeAndName decodes a key in type[/name] format into type and fullName.
 // fullName is the key normalized such that type and name components have spaces trimmed.
 // The "type" part must be present, the forward slash and "name" are optional.
-func decodeTypeAndName(key string) (typeStr, fullName string, err error) {
+func DecodeTypeAndName(key string) (configmodels.NamedEntity, error) {
```

Comment:
NamedEntity seems like only an approximate fit for what this function needs to return. For example DecodeTypeAndName does not need to return something that has an "IsEnabled()" capability.

I think it is better not to touch NamedEntity (which has a different purpose currently) and if you want to avoid returning a pair of strings just introduce a `TypeAndName` struct here and return it, e.g.:

```go
type TypeAndName struct {
    Type string
    Name string
}
```
I don't think this particular fixes in any way the actual problem that you found: the bug where `err` was not provably non-nil from caller's perspective (but it is provably non-nil if you examine this function).